src/components/Avatar/Avatar.tsx

```diff
@@ -2,14 +2,56 @@ import * as PropTypes from 'prop-types'
 import * as React from 'react'
 import { Image, Label, Icon } from '../../'
 
-import { customPropTypes, UIComponent } from '../../lib'
+import { customPropTypes, UIComponent, NotStrictProps } from '../../lib'
+import { ComponentVariablesInput, IComponentPartStylesInput } from '../../../types/theme'
+
+export interface IAvatarPropsStrict {
+  /** The alternative text for the image used in the Avatar. */
```

Comment:
a bit concerned whether we should duplicate this descriptions here now. The problem is that it will bring significant maintainability burden, as with each change to API we will need to ensure the following things:
- examples are consistent
- descriptions in prop types are valid
**- description of prop types correspond to these ones**
- changes in default props are properly reflected
- handled props are properly provided

Although there is quite a lot of burden already, won't like to make the situation even worse - as it seems that there is no huge benefit in these descriptions at this stage (while I do understand that IDE suggestions will be helpful, at this point we are rather interested in ensuring type correctness).

At the same time, I am pretty sure that we will be able to address this problem (of missing comments for interface properties) later, so that there will be always single source of truth for this information.

Please, let me know about your thoughts on that

Reply:
I agree with the points above, will remove the comments on the interface as we already have the same in the prop types.

Reply:
great - lets return to this question later and figure out generation mechanism for them :)